### PR TITLE
Implement unregisterAdvertisement

### PR DESCRIPTION
**Pull request by g-ortuno**

### CODE REVIEW

**Comment by g-ortuno** [comment]
@jyasskin: PTAL

File: libraries/eddystone-advertising-library/eddystone-advertising.js [hunk 1]
```diff
@@ -327,6 +327,26 @@
         });
       });
     }
+
+    /**
+       Function to unregister an advertisement.
+       @param {EddystoneAdvertisement} advertisement The advertisement to
+       unregister.
+       @returns {Promise} Which `fulfills` if the advertisement was unregistered
+       successfully, `rejects` with an `error` otherwise.
```
Comment (jyasskin) [inline libraries/eddystone-advertising-library/eddystone-advertising.js]: You should probably capitalize `Error` since it's the name of a type.

Comment (jyasskin) [inline libraries/eddystone-advertising-library/eddystone-advertising.js]: Mention what the developers should do here when the promise rejects. They have to get the user to reboot if they really want to make sure the advertisement goes away, right?

File: libraries/eddystone-advertising-library/eddystone-advertising.js [hunk 2]
```diff
@@ -327,6 +327,26 @@
         });
       });
     }
+
+    /**
+       Function to unregister an advertisement.
+       @param {EddystoneAdvertisement} advertisement The advertisement to
+       unregister.
+       @returns {Promise} Which `fulfills` if the advertisement was unregistered
```
Comment (jyasskin) [inline libraries/eddystone-advertising-library/eddystone-advertising.js]: Does `Promise<void>` work for the type?

File: libraries/eddystone-advertising-library/eddystone-advertising.md [hunk 3]
```diff
@@ -119,6 +119,7 @@ This is the object that holds the information about the registered BLE
   * [.type](#EddystoneAdvertisement+type) : <code>string</code>
   * [.url?](#EddystoneAdvertisement+url?) : <code>string</code>
   * [.txPower?](#EddystoneAdvertisement+txPower?) : <code>number</code>
```
Comment (jyasskin) [inline libraries/eddystone-advertising-library/eddystone-advertising.md]: I think I mentioned at some point that `txPower` should have a type of `number?`; the name of the field shouldn't be `txPower?`.

**Comment by jyasskin** [comment]
Otherwise, lgtm.

File: libraries/eddystone-advertising-library/eddystone-advertising.md (hunk 3 above)
Comment (g-ortuno) [inline libraries/eddystone-advertising-library/eddystone-advertising.md]: Adding the `?` to the type doesn't change the resulting documentation. That's why I left it in txPower.

File: libraries/eddystone-advertising-library/eddystone-advertising.js (hunk 2 above)
Comment (g-ortuno) [inline libraries/eddystone-advertising-library/eddystone-advertising.js]: Noup. The resulting text looks exactly the same.

File: libraries/eddystone-advertising-library/eddystone-advertising.js (hunk 1 above)
Comment (g-ortuno) [inline libraries/eddystone-advertising-library/eddystone-advertising.js]: Done.
Done.

**Comment by jyasskin** [comment]
LGTM. Too bad about the documentation generator.

**Comment by jyasskin** [comment]
Hm, does the syntax at http://usejsdoc.org/tags-type.html work for `Promise.<void>` and `?number`?

**Comment by g-ortuno** [comment]
`Promise.<void>` works but `?number` doesn't. Alternatively we could use something like:
`string | undefined`

**Comment by jyasskin** [comment]
Eh, I guess the type of the optional fields doesn't matter that much. I can live with either `txPower?` or `number | undefined`.

**Comment by g-ortuno** [comment]
Went with `number | undefined`

**Pull request merged**